kusto/query_builder.go

```diff
@@ -64,147 +65,146 @@ type ParamType struct {
 	name string
 }
 
-func (p ParamType) validate() error {
-	if !p.Type.Valid() {
-		return fmt.Errorf("the .Type was not a valid value, must be one of the values in this package starting with CT<type name>, was %s", p.Type)
-	}
-	if p.Default == nil {
-		return nil
+// isValid validates whether the value is of known kusto Types.
+func isValid(kustoType types.Column, val interface{}) bool {
+	if !kustoType.Valid() {
+		return false
 	}
 
-	switch p.Type {
+	switch kustoType {
 	case types.Bool:
-		if _, ok := p.Default.(bool); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
-		}
-		return nil
+		_, ok := val.(bool)
+		return ok
 	case types.DateTime:
-		if _, ok := p.Default.(time.Time); !ok {
-			return fmt.Errorf("the .Type was %s, but the value was a %T", p.Type, p.Default)
-		}
-		return nil
+		_, ok := val.(time.Time)
+		return ok
 	case types.Dynamic:
```

Comment:
This should return true, because any type could potentially be OK for dynamic. (that's why one of tests fail)